src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,193 @@
+package duke;
+
+import java.util.ArrayList;
+import java.util.stream.Stream;
+
+import duke.tasks.Deadline;
+import duke.tasks.Event;
+import duke.tasks.Task;
+import duke.tasks.ToDo;
+
+/**
+ * TaskList contains the list of Tasks and methods to modify the list.
+ */
+public class TaskList {
+    private ArrayList<Task> listOfTasks;
+    private ArrayList<Task> listOfFindTasks;
```

Comment:
Maybe this could have been
```suggestion
    private ArrayList<Task> listOfFoundTasks;
```
Since 'find' is a verb and using an adjective can make it less confusing